backup_preload_mongo.sh

```diff
@@ -0,0 +1,1673 @@
+    #TODO remove finalizers before deleting
+    oc delete pvc cs-mongodump -n $FROM_NAMESPACE --ignore-not-found --timeout=10s
+    if [ $? -ne 0 ]; then
+        info "Failed to delete pvc cs-mongodump, patching its finalizer to null..."
+        oc patch pvc cs-mongodump -n $FROM_NAMESPACE --type="json" -p '[{"op": "remove", "path":"/metadata/finalizers"}]'
+    fi
+  fi
+  pvcexists=$(oc get pvc cs-mongodump -n $TO_NAMESPACE --no-headers --ignore-not-found | awk '{print $2}')
+  if [[ -n "$pvcexists" ]]; then
+    if [[ "$pvcexists" == "Bound" ]]; then
+      dv=$(oc get pvc cs-mongodump -n $TO_NAMESPACE -o=jsonpath='{.spec.volumeName}')
+      oc patch pv $dv -p '{"spec": { "persistentVolumeReclaimPolicy" : "Delete" }}'
+    fi
+    oc delete pvc cs-mongodump -n $TO_NAMESPACE --ignore-not-found --timeout=10s
+    if [ $? -ne 0 ]; then
+        info "Failed to delete pvc cs-mongodump, patching its finalizer to null..."
+        oc patch pvc cs-mongodump -n $TO_NAMESPACE --type="json" -p '[{"op": "remove", "path":"/metadata/finalizers"}]'
+    fi
+  fi
+  success "Previous run cleaned up."
+} # cleanup
+
+
+#
+#  Create the dump PVC
+#
+function createdumppvc() {
+  title "Creating a PVC for the MongoDB dump"
+  msg "-----------------------------------------------------------------------"
+  oc project $FROM_NAMESPACE
+  currentns=$(oc project -q)
+  if [[ "$currentns" -ne "$FROM_NAMESPACE" ]]; then
+    error "Cannot switch to $FROM_NAMESPACE"
+  fi
+
+  stgclass=$(oc get pvc mongodbdir-icp-mongodb-0 -o=jsonpath='{.spec.storageClassName}')
+  if [[ -z $stgclass ]]; then
+    error "Cannnot get storage class name from PVC mongodbdir-icp-mongodb-0 in $FROM_NAMESPACE"
+  fi
+
+  cat <<EOF >$TEMPFILE
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: cs-mongodump
+  namespace: $FROM_NAMESPACE
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 20Gi
+  storageClassName: $stgclass
+  volumeMode: Filesystem
+EOF
+
+  oc apply -f $TEMPFILE
+
+  status=$(oc get pvc cs-mongodump --no-headers | awk '{print $2}')
+  while [[ "$status" != "Bound" ]]
+  do
+    info "Waiting for pvc cs-mongodump to bind"
+    sleep 10
+    status=$(oc get pvc cs-mongodump --no-headers | awk '{print $2}')
+  done
+  success "MongoDB backup ready"
+
+} # createdumppvc
+
+
+#
+# Backup(Dump) the mongodb in the from: namespace
+#
+function dumpmongo() {
+  title "Backing up MongoDB in namespace $FROM_NAMESPACE"
+  msg "-----------------------------------------------------------------------"
+  currentns=$(oc project $FROM_NAMESPACE -q)
+  if [[ "$currentns" -ne "$FROM_NAMESPACE" ]]; then
+    error "Cannot switch to $FROM_NAMESPACE"
+  fi
+
+  cat <<EOF >$TEMPFILE
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: mongodb-backup
+spec:
+  parallelism: 1
+  completions: 1
+  backoffLimit: 20
+  template:
+    spec:
+      containers:
+      - name: cs-mongodb-backup
+        image: quay.io/opencloudio/ibm-mongodb:4.0.24
```

Comment:
We need to use the image which matches CP2 image

Reply:
This image is for the mongo backup job, would it be mirrored?